### PR TITLE
fix (TDI-37028): salesforce doesn't support write BigDecimal

### PR DESCRIPTION
**Pull request by jzhao-talend**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
We upgrade salesforce and wsc in master, but not for product 6.2.
The wsc which we used didn't support BigDecimal. So we need to back port the changes of  salesforce and wsc to  components 0.13

**What is the new behavior?**
1) upgrade salesforce and wsc in 0.13
2) add junit

**Does this PR introduce a breaking change?**

- [x] No

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/396/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/396/)

File: components-osgitest/pom.xml [hunk 1]
```diff
@@ -75,7 +75,12 @@
 			<version>${salesforceversion}</version>
 			<scope>provided</scope>
 		</dependency>
-
+		<!-- required for salesforce depenencies -->
+		<dependency>
```
Comment (gonchar-ivan) [inline components-osgitest/pom.xml]: Maybe this dependency could go to Salesforce's pom.xml?

Comment (sgandon) [inline components-osgitest/pom.xml]: indeed this bundle should never be here.

**Comment by jzhao-talend** [comment]
@sgandon
yes. you are right. I am only cherry-pick your commit https://github.com/Talend/components/commit/71e5788c8767c7ab74f929844a582505e6016ec5
but I forgot that  you have made osgi tests work individually for each component in this commit
https://github.com/Talend/components/commit/f23e7d7cc459d86bca42c8e19264768908cae96d
I would remove it from from sogi-test.

Thanks Ivan also.

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/423/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/423/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/437/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/437/)

**Comment by sgandon** [comment]
you can merge it.

**Pull request merged**